### DISCUSSION TITLE
Get amount of files with violations and violations per file

### DISCUSSION THREAD

**Discussion by mjhymowitz** [rubocop/rubocop#11911] [Ideas]
Hi, My team recently implemented rubocop into our project and it has a lot of violations. I want to work on the issue, file by file. I'm trying to get some more statistics data from rubocop to break down the problem but there doesn't seem to be a way. Currently I can only see that out of the 3K files in the project there are 10K violations.

My Idea is to add a command that returns a list of file names and the amount of violations in each file(`rubocop -statisics`). Maybe add a way to get just the top X amount of offending files (`rubocop -statisics X`)

**Comment by koic** [comment]
The Worst Offenders Formatter is provided for that purpose:

```console
$ rubocop --format worst

89  this/file/is/really/bad.rb
2   much/better.rb
--
91  Total in 2 files
```

To display only the top results, it can be filtered using the `head` command:

```console
$ rubocop --format worst | head -n 42
```

https://docs.rubocop.org/rubocop/1.65/formatters.html#worst-offenders-formatter
